pkg/types/codec.go

```diff
@@ -4,6 +4,13 @@ import (
 	"context"
 )
 
+const (
+	ErrFieldNotFound         = InvalidArgumentError("field not found")
+	ErrInvalidEncoding       = InvalidArgumentError("invalid encoding")
+	ErrWrongNumberOfElements = InvalidArgumentError("wrong number of elements in slice")
```

Comment:
Would it be correct/equivalent to say 'length' instead of 'number of elements':
```suggestion
	ErrWrongLength = InvalidArgumentError("slice is wrong length")
```
If so, I think some version of this should be preferred to avoid introducing new language.

(and should name contain 'slice'? e.g. `ErrSliceWrongLength`)